sklearn/neighbors/_binary_tree.pxi

```diff
@@ -986,7 +986,7 @@ cdef class BinaryTree:
 
         # Allocate tree-specific data
         allocate_data(self, self.n_nodes, n_features)
-        self._recursive_build(0, 0, n_samples)
+        self._recursive_build(self.node_data_arr, 0, 0, n_samples)
```

Comment:
Since this line gets modified:
```suggestion
        self._recursive_build(
            node_data=self.node_data_arr,
            i_node=0,
            idx_start=0,
            idx_end=n_samples,
        )
```